asllib/doc/ASLSemanticsReference.tex

```diff
@@ -967,95 +971,87 @@ \chapter{Evaluation of Expressions \label{chap:eval_expr}}
-  \vg \eqdef \emptygraph \\
-  \newenv \eqdef \env}
-  { \evalexpr{\env, \ELiteral(\vv)} \evalarrow \Normal((\vres,\vg), \env) }
+\inferrule{}{
+  \evalexpr{\env, \overname{\ELiteral(\vl)}{\ve}} \evalarrow \Normal((\overname{\nvliteral{\vl}}{\vv},\overname{\emptygraph}{\vg}), \overname{\env}{\newenv})
+}
 \end{mathpar}
 \end{emptyformal}
 
 \isempty{\subsection{Comments}}
 
 \section{SemanticsRule.ELocalVar \label{sec:SemanticsRule.ELocalVar}}
-  \subsection{Prose}
-  Evaluation of the expression $\ve$ in an environment $\env$ is
-  $\Normal((\vv, \vg), \newenv)$ and all of the following apply:
-  Evaluation of the expression $\ve$ in an environment $\env$ is
-  $\Normal((\vv, \vg), \newenv)$ and all of the following apply:
-  \begin{itemize}
-    \item $\ve$ denotes a variable expression, $\EVar(\vx)$,
-    which is bound locally in $\env$;
-    \item $\vv$ is the value of $\vx$ in $\env$;
-    \item $\newenv$ is $\env$.
-    \item $\vg$ is the graph containing a single Read Effect for $\vx$.
-  \end{itemize}
+\subsection{Prose}
+All of the following apply:
+\begin{itemize}
+  \item $\ve$ denotes a variable expression, $\EVar(\vx)$,
+        which is bound locally in $\env$;
+  \item $\vv$ is the value of $\vx$ in $\env$;
+  \item $\newenv$ is $\env$.
+  \item $\vg$ is the graph containing a single Read Effect for $\vx$.
+\end{itemize}
 
-  \subsection{Example}
-    In the specification:
-    \VerbatimInput{../tests/ASLSemanticsReference.t/SemanticsRule.ELocalVar.asl}
-    the evaluation of \texttt{x} within \texttt{assert x == 3;} uses SemanticsRule.ELocalVar.
+\subsection{Example}
+In the specification:
+\VerbatimInput{../tests/ASLSemanticsReference.t/SemanticsRule.ELocalVar.asl}
+the evaluation of \texttt{x} within \texttt{assert x == 3;} uses SemanticsRule.ELocalVar.
 
-  \CodeSubsection{\ELocalVarBegin}{\ELocalVarEnd}{../Interpreter.ml}
+\CodeSubsection{\ELocalVarBegin}{\ELocalVarEnd}{../Interpreter.ml}
 
 \begin{emptyformal}
-  \subsection{Formally}
-  \begin{mathpar}
-  \inferrule{
-    \env \eqname (\tenv, \denv)\\
-    \vx \in \dom(L^\denv)\\
-    \vv \eqdef L^\denv(\vx) \\
-    \vg \eqdef \ReadEffect(\vx)\\
-    \newenv \eqname \env
-  }
-  {\evalexpr{\env, \EVar(\vx)} \evalarrow \Normal((\vv, \vg), \newenv)}
-  \end{mathpar}
+\subsection{Formally}
+\begin{mathpar}
+\inferrule{
+  \env \eqname (\Ignore, \denv)\\
+  \vx \in \dom(L^\denv)
+}{
+  \evalexpr{\env, \EVar(\vx)} \evalarrow \Normal((\overname{L^\denv(\vx)}{\vv}, \overname{\ReadEffect(\vx)}{\vg}), \overname{\env}{\newenv})
+}
+\end{mathpar}
 \end{emptyformal}
 
-  \subsection{Comments}
-  When there exists a global variable $\vx$, the type system
-  forbids having $\vx$ as a local variable.
-  This is enforced by TypingRule.LDVar and TypingRule.DeclareOneFunc.
+\subsection{Comments}
+When there exists a global variable $\vx$, the type system
+forbids having $\vx$ as a local variable.
+This is enforced by TypingRule.LDVar and TypingRule.DeclareOneFunc.
```

Comment:
please add a reference to the Typing Reference, with pointers to the specific chapters